src/JsonApiDotNetCore.MongoDb/MongoDbRepository.cs

```diff
@@ -0,0 +1,230 @@
+
+            var expression = builder.ApplyQuery(layer);
+            return (IMongoQueryable<TResource>)source.Provider.CreateQuery<TResource>(expression);
+        }
+        
+        protected virtual IQueryable<TResource> GetAll()
+        {
+            return Collection.AsQueryable();
+        }
+
+        /// <inheritdoc />
+        public virtual Task<TResource> GetForCreateAsync(TId id, CancellationToken cancellationToken)
+        {
+            var resource = _resourceFactory.CreateInstance<TResource>();
+            resource.Id = id;
+
+            return Task.FromResult(resource);
+        }
+
+        /// <inheritdoc />
+        public virtual Task CreateAsync(TResource resourceFromRequest, TResource resourceForDatabase,
+            CancellationToken cancellationToken)
+        {
+            if (resourceFromRequest == null) throw new ArgumentNullException(nameof(resourceFromRequest));
+            if (resourceForDatabase == null) throw new ArgumentNullException(nameof(resourceForDatabase));
+            
+            foreach (var attribute in _targetedFields.Attributes)
+            {
+                attribute.SetValue(resourceForDatabase, attribute.GetValue(resourceFromRequest));
+            }
+
+            return Collection.InsertOneAsync(resourceForDatabase, new InsertOneOptions(), cancellationToken);
+        }
+
+        /// <inheritdoc />
+        public virtual async Task<TResource> GetForUpdateAsync(QueryLayer queryLayer, CancellationToken cancellationToken)
+        {
+            var resources = await GetAsync(queryLayer, cancellationToken);
+            return resources.FirstOrDefault();
+        }
+
+        /// <inheritdoc />
+        public virtual async Task UpdateAsync(TResource resourceFromRequest, TResource resourceFromDatabase, CancellationToken cancellationToken)
+        {
+            if (resourceFromRequest == null) throw new ArgumentNullException(nameof(resourceFromRequest));
+            if (resourceFromDatabase == null) throw new ArgumentNullException(nameof(resourceFromDatabase));
+            
+            foreach (var attr in _targetedFields.Attributes)
+                attr.SetValue(resourceFromDatabase, attr.GetValue(resourceFromRequest));
+
+            await Collection.ReplaceOneAsync(
+                Builders<TResource>.Filter.Eq(e => e.Id, resourceFromDatabase.Id),
+                resourceFromDatabase,
+                new ReplaceOptions(),
+                cancellationToken);
+        }
+
+        /// <inheritdoc />
+        public virtual async Task DeleteAsync(TId id, CancellationToken cancellationToken)
+        {
+            var result = await Collection.DeleteOneAsync(
+                Builders<TResource>.Filter.Eq(e => e.Id, id),
+                new DeleteOptions(),
+                cancellationToken);
+
+            if (!result.IsAcknowledged)
+            {
+                throw new DataStoreUpdateException(new Exception("Delete operation was not acknowledged by MongoDB"));
```

Comment:
Sentences should end with a dot. And it would be nice to include the `id` value here, too. Suggestion:
```
throw new DataStoreUpdateException(new Exception($"Failed to delete document with id '{id}', because the operation was not acknowledged by MongoDB."));
```